csharp/SecureMemory/SecureMemory/SecureMemoryImpl/Linux/LinuxSecureMemoryAllocatorLP64.cs

```diff
@@ -0,0 +1,104 @@
+using System;
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+using GoDaddy.Asherah.PlatformNative.LP64.Linux;
+using GoDaddy.Asherah.PlatformNative.LP64.Linux.Enums;
+using GoDaddy.Asherah.SecureMemory.SecureMemoryImpl.Libc;
+
+[assembly: InternalsVisibleTo("SecureMemory.Tests")]
+[assembly: InternalsVisibleTo("DynamicProxyGenAssembly2")]
+
+namespace GoDaddy.Asherah.SecureMemory.SecureMemoryImpl.Linux
+{
+    /*
+     * Linux protected memory implementation supports:
+     *
+     * mmap(MAP_PRIVATE | MAP_ANON) - Private anonymous
+     * madvise(MADV_DONTDUMP) - Selective core dump avoidance
+     */
+
+    // ReSharper disable once InconsistentNaming
+    internal class LinuxSecureMemoryAllocatorLP64 : LibcSecureMemoryAllocatorLP64
+    {
+        private readonly int pageSize = Environment.SystemPageSize;
+        private readonly LinuxLibcLP64 libc;
+
+        public LinuxSecureMemoryAllocatorLP64()
+            : base(new LinuxLibcLP64())
+        {
+            Debug.WriteLine("LinuxSecureMemoryAllocatorLP64 ctor");
+            libc = (LinuxLibcLP64)GetLibc();
+        }
+
+        public LinuxSecureMemoryAllocatorLP64(LinuxLibcLP64 libc)
+            : base(libc)
+        {
+            this.libc = libc;
+        }
+
+        public override void Dispose()
+        {
+        }
+
+        internal override int GetRlimitCoreResource()
+        {
+            return (int)RlimitResource.RLIMIT_CORE;
```

Comment:
If I understand correctly, the protected memory allocator version locks, but the secure memory allocator does not.  If that's the case, we can/should probably drop the rlimit and locking related aspects out of the secure memory allocator right?

Reply:
Right, but `RLIMIT_CORE` is used to restrict the size of core dump files (which we use to disable core dumps), so this one doesn't really fall into the "rlimit and locking" category.